diff --git a/dist/runtime/server/api.js b/dist/runtime/server/api.js
index 3685dcb9e566d70d675d6557c95106dddff1d6d2..22da932a0afd12f9dfd53f8cd674205ec6779f4c 100644
--- a/dist/runtime/server/api.js
+++ b/dist/runtime/server/api.js
@@ -1,9 +1,10 @@
 import { getIcons } from "@iconify/utils";
 import { hash } from "ohash";
-import { createError } from "h3";
+import { HTTPError } from "nitro/h3";
+import { defineCachedHandler } from "nitro/cache";
 import { parseQuery, parsePath } from "ufo";
 import { consola } from "consola";
-import { useAppConfig, defineCachedEventHandler } from "#imports";
+import { useAppConfig } from "#imports";
 import { collections } from "#nuxt-icon-server-bundle";
 const warnOnceSet = /* @__PURE__ */ new Set();
 const DEFAULT_ENDPOINT = "https://api.iconify.design";
@@ -14,14 +15,14 @@ function getInstallCommand(pkg) {
   if (ua.startsWith("bun")) return `bun add -D ${pkg}`;
   return `npm i -D ${pkg}`;
 }
-export default defineCachedEventHandler(async (event) => {
+export default defineCachedHandler(async (event) => {
   const options = useAppConfig().icon;
   const collectionName = event.context.params?.collection?.replace(/\.json$/, "");
   const collection = collectionName && Object.hasOwn(collections, collectionName) ? await collections[collectionName]?.() : null;
   const apiEndPoint = options.iconifyApiEndpoint || DEFAULT_ENDPOINT;
-  const icons = String(parseQuery(parsePath(event.path).search).icons || "").split(",");
-  if (!collectionName) return createError({ status: 400, message: "No collection specified" });
-  if (!icons.length) return createError({ status: 400, message: "No icons specified" });
+  const icons = String(parseQuery(parsePath(event.url.pathname + event.url.search).search).icons || "").split(",");
+  if (!collectionName) return new HTTPError({ status: 400, message: "No collection specified" });
+  if (!icons.length) return new HTTPError({ status: 400, message: "No icons specified" });
   if (!collection && import.meta.dev && !warnOnceSet.has(collectionName) && apiEndPoint === DEFAULT_ENDPOINT) {
     consola.warn([
       `[Icon] Collection \`${collectionName}\` is not found locally`,
@@ -41,7 +42,7 @@ export default defineCachedEventHandler(async (event) => {
     const apiUrl = new URL(`./${collectionName}.json?icons=${icons.join(",")}`, apiEndPoint);
     consola.debug(`[Icon] fetching ${icons.map((i) => "`" + collectionName + ":" + i + "`").join(",")} from iconify api`);
     if (apiUrl.host !== new URL(apiEndPoint).host) {
-      return createError({ status: 400, message: "Invalid icon request" });
+      return new HTTPError({ status: 400, message: "Invalid icon request" });
     }
     try {
       const data = await $fetch(apiUrl.href);
@@ -49,18 +50,18 @@ export default defineCachedEventHandler(async (event) => {
     } catch (e) {
       consola.error(e);
       if (e.status === 404)
-        return createError({ status: 404 });
+        return new HTTPError({ status: 404 });
       else
-        return createError({ status: 500, message: "Failed to fetch fallback icon" });
+        return new HTTPError({ status: 500, message: "Failed to fetch fallback icon" });
     }
   }
-  return createError({ status: 404 });
+  return new HTTPError({ status: 404 });
 }, {
   group: "nuxt",
   name: "icon",
   getKey(event) {
     const collection = event.context.params?.collection?.replace(/\.json$/, "") || "unknown";
-    const icons = String(parseQuery(parsePath(event.path).search).icons || "").split(",");
+    const icons = String(parseQuery(parsePath(event.url.pathname + event.url.search).search).icons || "").split(",");
     return `${collection}_${icons[0]}_${icons.length}_${hash(icons.join(","))}`;
   },
   swr: true,
